x-pack/legacy/plugins/reporting/server/browsers/chromium/driver/chromium_driver.ts

```diff
@@ -70,6 +83,31 @@ export class HeadlessChromiumDriver {
       interceptedCount = interceptedCount + (isData ? 0 : 1);
     });
 
+    this.page.on('response', interceptedResponse => {
+      const interceptedUrl = interceptedResponse.url();
+      const remoteIp = interceptedResponse.remoteAddress().ip;
+      const status = interceptedResponse.status();
+      const isMalicious = interceptedUrl.startsWith('file://');
+      let allowed = true;
+
+      // Redirects can come through this hook as well, so only
+      // filter payloads that actually respond
+      if (this.networkPolicy.enabled && status === 200) {
```

Comment:
The networkPolicy is only being enforced on the response, is this because we have the remote address's ip address at this point in-time? It feels like we'd also want to block requests based on the protocol and hostname during the request itself. Otherwise, we aren't stopping the request from occurring, and this could be used to POST sensitive information to an endpoint an attacker controls.

Reply:
Yes, unfortunately we don't have a sense of the remote IP prior to making the request. I think I can make the utility flexible enough to use in both scenarios